pyvista/core/composite.py

```diff
@@ -690,3 +703,238 @@ def copy(self, deep=True):
+
+                if field != FieldAssociation.NONE:
+                    data_assoc.append((field, scalars, block))
+
+        if name is None:
+            return FieldAssociation.NONE, np.array([])
+
+        if not data_assoc:
+            raise KeyError(f'"{name}" is missing from all the blocks of this composite dataset.')
+
+        # Verify that all the data has identical field association
+        first_field = data_assoc[0][0]
+        if any([first_field != field for field, _, _ in data_assoc]):
+            # remove any data not matching the preference
+            for field, scalars, block in list(data_assoc):
+                if field.name.lower() != preference:
+                    block.set_active_scalars(None)
+                    data_assoc.remove((field, scalars, block))
+
+            if preference == 'point':
+                field_asc = FieldAssociation.POINT
+            else:
+                field_asc = FieldAssociation.CELL
+
+        else:
+            field_asc = first_field
+
+        # Verify array consistency
+        dims: Set[int] = set()
+        dtypes: Set[np.dtype] = set()
+        for block in self:
+            for _, scalars, _ in data_assoc:
+                dims.add(scalars.ndim)
+                dtypes.add(scalars.dtype)
+
+        if len(dims) > 1:
+            raise ValueError(f'Inconsistent dimensions {dims} in active scalars.')
+
+        # check complex mismatch
+        is_complex = [np.issubdtype(dtype, np.complexfloating) for dtype in dtypes]
+        if any(is_complex) and not all(is_complex):
+            raise ValueError('Inconsistent complex and real data types in active scalars.')
+
+        return field_asc, scalars
+
+    def as_polydata(self):
+        """Convert all the datasets in this MultiBlock to :class:`pyvista.PolyData`.
+
+        This will return a new `pyvista.MultiBlock` dataset. Blocks that are
+        already :class:`pyvista.PolyData` will not be copied.
+
+        Returns
+        --------
+        pyvista.MultiBlock
+            MultiBlock containing only PolyData datasets.
+
+        """
+        # we make a shallow copy here to avoid modifying the original dataset
+        dataset = self.copy(deep=False)
+
+        # always convert to polydata
+        for i, block in enumerate(dataset):
+            if block is not None:
+                if isinstance(block, MultiBlock):
+                    dataset[i] = block.as_polydata()
+                elif not isinstance(block, pyvista.PolyData):
+                    dataset[i] = block.extract_surface()
+                else:
+                    # guarantee that the new instance is a shallow copy
+                    dataset[i] = block.copy(deep=False)
+            else:
+                # must have empty polydata within these datasets to ensure
+                # downstream filters don't work on null pointers
+                dataset[i] = pyvista.PolyData()
```

Comment:
It may be worth pointing out that null blocks will be non-null in the docstring

Reply:
Done in 416ef55af169ce2da8d4443e3c1d17eeff9d7364.